client/clientfactory.go

```diff
@@ -177,6 +177,7 @@ func (cf *rpcClientFactory) NewMatchingClientWithTimeout(
 		longPollTimeout,
 		common.NewClientCache(keyResolver, clientProvider),
 		matching.NewLoadBalancer(namespaceIDToName, cf.dynConfig),
+		cf.dynConfig.GetBoolProperty(dynamicconfig.MatchingUseOldRouting, false),
```

Comment:
The default value should be true so during upgrade, there is no disruption and after upgrade one time dynamic config change is needed to switch on new routing. In v1.19 release we would remove this dynamic config and change the behavior to new routing. 
If you use default value of false, user has to change dynamic config twice to have smooth transition.

Reply:
1. Unless they want to keep a useless value in dynamic config forever, everyone needs to do two changes anyway.
2. I wanted to make it possible to "complete" the upgrade process after 1.18 and not have to keep an action item for 1.19 (to remove the useless value)
3. Many users are going to not read release notes and just upgrade. I would rather they hit this (and get the benefits of this) on the 1.18 upgrade than on 1.19

Reply:
It is likely there is already bunch of deprecated dynamic config keys exists in their config files.  And they can cleanup useless dynamic config keys anytime they want, but it is not required. 

You still need to remove this dynamic config key in 1.19. 

The benefit is only visible when you have large number of namespaces, for most self hosted environment, that is not the case.

Reply:
Ok, I changed it